tests/ExtensibleConf.hs

```diff
@@ -0,0 +1,25 @@
+{-# OPTIONS_GHC -Wall #-}
+module ExtensibleConf where
+
+import Test.Hspec
+
+import XMonad
+import qualified XMonad.Util.ExtensibleConf as XC
+
+spec :: Spec
+spec = do
+    specify "lookup" $
+        XC.lookup def `shouldBe` (Nothing :: Maybe ())
+    specify "lookup . add" $
+        XC.lookup (XC.add "a" def) `shouldBe` Just "a"
+    specify "lookup . add . add" $
+        XC.lookup (XC.add "b" (XC.add "a" def)) `shouldBe` Just "ab"
+
```

Comment:
I added a similar test: feb80864

I can't decide about the rest. :-/
Some people seem to have strong opinions, but I don't. I wish it was possible to write `once` without messing with the types. Then I'd just say that the code won't be modified anyway and push whatever a dice roll tells me to push. But as this abstraction leaks into higher levels, it may be worth debating a little bit. Just a little bit though… ;-)

Reply:
Well, I suppose my opinion should be obvious—I think the version with `TypeApplications` is much more readable.  I will, however, gladly open this up to some productive bike-shedding :>

Reply:
So how about this:

I pushed ee35ad65c6b3a97cff829915cc76eb05a087f22c, which uses TypeApplications, but doesn't define `theType` (if Haskell gods wanted us to have `forall a. Typeable a => TypeRep`, they'd have put it in `base`). If people oppose that impl, they're free to comment constructively here.

Reply:
Works for me :+1:

>(if Haskell gods wanted us to have `forall a. Typeable a => TypeRep`, they'd have put it in `base`)

One could argue that that's just a mistake in `base` :P